src/support.cpp

```diff
@@ -783,6 +790,215 @@ void AreaSupport::generateOverhangAreasForMesh(SliceDataStorage& storage, SliceM
+        {
+            if (edge.is_infinite())
+            {
+                continue;
+            }
+
+            auto p0 = VoronoiUtils::p(edge.vertex0());
+            auto p1 = VoronoiUtils::p(edge.vertex1());
+
+            // skip edges that move "outside" the polygon;
+            // these are st edges that are inside polygon-holes
+            if (! layer_delta.inside(p0) && ! layer_delta.inside(p1))
+            {
+                continue;
+            }
+
+            auto dist_to_center_edge = static_cast<double>(cura::vSize(p0 - p1));
+
+            if (dist_to_center_edge < snap_radius)
+            {
+                continue;
+            }
+
+            // p0 to p1 is the distance to the center between the two polygons; two times
+            // this distance is (approximately) the distance between the boundaries
+            auto dist_to_boundary = 2. * dist_to_center_edge;
+            auto slope = dist_to_boundary / delta_z;
+
+            auto nearby_vals = slope_at_point.getNearbyVals(p0, search_radius);
+            auto n = ranges::accumulate(nearby_vals | views::get( &point_pair_t::first ), 0);
+            auto cumulative_slope = ranges::accumulate(nearby_vals | views::get( &point_pair_t::second ), 0.);
+
+            n += 1;
+            cumulative_slope += slope;
+
+            // update cumulative_slope in sparse grid
+            slope_at_point.insert(p0, { n, cumulative_slope });
+        }
+
+        for (const auto& poly: layer_current)
+        {
+            for (const auto& point: poly)
+            {
+                auto nearby_vals = slope_at_point.getNearbyVals(point, search_radius);
+                auto n = ranges::accumulate(nearby_vals | views::get( &point_pair_t::first ), 0);
+                auto cumulative_slope = ranges::accumulate(nearby_vals | views::get( &point_pair_t::second ), 0.);
+
+                if (n != 0)
+                {
+                    auto slope = cumulative_slope / static_cast<double>(n);
+                    auto wall_angle = std::atan(slope);
+                    auto ratio = std::min(wall_angle / overhang_angle, 1.);
+
+                    auto xy_distance_varying = std::lerp(xy_distance, xy_distance_natural, ratio);
+
+                    auto nearby_vals_offset_dist = offset_dist_at_point.getNearbyVals(point, search_radius);
+
+                    // update and insert cumulative varying xy distance in one go
+                    offset_dist_at_point.insert(point, {
+                                                       ranges::accumulate(nearby_vals_offset_dist | views::get( &point_pair_t::first ), 0) + 1,
+                                                       ranges::accumulate(nearby_vals_offset_dist | views::get( &point_pair_t::second ), 0.) + xy_distance_varying
+                                                   });
+                }
+            }
+        }
+    }
+
+    std::vector<int> varying_offsets;
```

Comment:
```suggestion
    std::vector<coord_t> varying_offsets;
```

`coord_t` is `long long int` otherwise we might get narrowing cast warnings somewhere